packages/flutter/lib/src/widgets/image_filter.dart

```diff
@@ -11,6 +11,12 @@ import 'framework.dart';
 
 /// Applies an [ImageFilter] to its child.
 ///
+/// An image filter will always apply its filter operation to the child widget,
+/// even if said filter is conceptually a "no-op", such as an ImageFilter.blur
+/// with a radius of 0 or an ImageFilter.matrix with an identity matrix. Setting
```

Comment:
Checking whether an image filter is a no-op and doing this optimization automatically is not an option?

Reply:
Consider the case where you are animating a transform using ImageFilter.matrix. At one point, your animation may cause the matrx to pass through the identity. Dropping and re-creating the image filter would tank performance (relatively) with no way for the user to control it.

Now consider if we special cased other filters, we'd have inconsistent behavior on how well "optimized" certain filters are.

Keeping it consistent seems like the best approach.

Reply:
It could be worthwhile to add some of that information to the docs here for future readers who are wondering the same.

Reply:
Updated, PTAL